appmap/test/test_detect_enabled.py

```diff
@@ -0,0 +1,143 @@
+    def test_some__recording_method_enabled(self):
+        for recording_method in RECORDING_METHODS:
+            DetectEnabled.initialize(recording_method)
+            assert DetectEnabled.should_enable() == True
+
+    recording_methods_as_false = {
+        '_'.join([ "APPMAP", "RECORD", recording_method.upper()]): "false"
+        for recording_method in RECORDING_METHODS
+    }
+    @patch.dict(os.environ, recording_methods_as_false)
+    def test_some__recording_method_disabled(self):
+        for recording_method in RECORDING_METHODS:
+            DetectEnabled.initialize(recording_method)
+            assert DetectEnabled.should_enable() == False
+
+    # Flask
+    def test_flask__none(self):
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == False
+
+    @patch.dict(os.environ, {"FLASK_APP": "app.py"})
+    def test_flask__flask_app(self):
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == False
+
+    @patch.dict(os.environ, {"FLASK_DEBUG": "0"})
+    def test_flask__flask_debug_0(self):
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == False
+
+    @patch.dict(os.environ, {"FLASK_DEBUG": "1"})
+    def test_flask__flask_debug_1(self):
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == True
+
+    @patch.dict(os.environ, {"FLASK_ENV": "production"})
+    def test_flask__flask_env_production(self):
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == False
+
+    @patch.dict(os.environ, {"FLASK_ENV": "development"})
+    def test_flask__flask_env_development(self):
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == True
+
+    # Django
+    def test_django__none(self):
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == False
+
+    @staticmethod
+    def settings_most():
+        return """
+# If the SECRET_KEY isn't defined we get the misleading error message
+# CommandError: You must set settings.ALLOWED_HOSTS if DEBUG is False.
+SECRET_KEY = "3*+d^_kjnr2gz)4q2m(&&^%$p4fj5dk3%lz4pl3g4m-%6!gf&)"
+
+# Must set ROOT_URLCONF else we get
+# AttributeError: 'Settings' object has no attribute 'ROOT_URLCONF'
+ROOT_URLCONF = "app.urls"
+
+MIDDLEWARE = ["appmap.django.Middleware"]
+
+"""
+
+    @patch.dict(os.environ, {"DJANGO_SETTINGS_MODULE": "app.settings_debug_false"})
+    def test_django__debug_false(self, data_dir, monkeypatch):
+        os.environ.setdefault("PYTHONPATH", os.getcwd())
+        filename = data_dir / "django" / "app" / "settings_debug_false.py"
+        file_write(str(filename), TestDetectEnabled.settings_most() + """DEBUG = False
+""")
+        monkeypatch.chdir(str(data_dir / "django"))
+        DetectEnabled.initialize(None)
+        assert DetectEnabled.should_enable() == False
+        file_delete(filename)
```

Comment:
This will leave behind the file if the `assert` fails, so you could do this to ensure it gets cleaned up.

```suggestion
        try:
            monkeypatch.chdir(str(data_dir / "django"))
            DetectEnabled.initialize(None)
            assert DetectEnabled.should_enable() == False
        finally:
            file_delete(filename)
```